File: ios/CinnaBookkeeping/Domain/Categories.swift
import Foundation

enum CategoryCatalog {
    static let expenses: [BookkeepingCategory] = [
        .init(kind: .expense, name: "餐饮", emoji: "🍜"),
        .init(kind: .expense, name: "交通", emoji: "🚌"),
        .init(kind: .expense, name: "购物", emoji: "🛍️"),
        .init(kind: .expense, name: "娱乐", emoji: "🎮"),
        .init(kind: .expense, name: "住房", emoji: "🏠"),
        .init(kind: .expense, name: "医疗", emoji: "💊"),
        .init(kind: .expense, name: "教育", emoji: "📚"),
        .init(kind: .expense, name: "运动", emoji: "🏃"),
        .init(kind: .expense, name: "旅行", emoji: "✈️"),
        .init(kind: .expense, name: "日用", emoji: "🧴"),
        .init(kind: .expense, name: "宠物", emoji: "🐾"),
        .init(kind: .expense, name: "其他", emoji: "📦")
    ]

    static let incomes: [BookkeepingCategory] = [
        .init(kind: .income, name: "工资", emoji: "💼"),
        .init(kind: .income, name: "奖金", emoji: "🎁"),
        .init(kind: .income, name: "副业", emoji: "💡"),
        .init(kind: .income, name: "理财", emoji: "📈"),
        .init(kind: .income, name: "红包", emoji: "🧧"),
        .init(kind: .income, name: "其他", emoji: "📦")
    ]

    static func categories(for kind: RecordKind) -> [BookkeepingCategory] {
        kind == .expense ? expenses : incomes
    }

    static func emoji(for category: String) -> String {
        (expenses + incomes).first(where: { $0.name == category })?.emoji ?? "📦"
    }
}

File: android/app/src/main/java/com/icelemon233/cinnabookkeeping/domain/Categories.kt
package com.icelemon233.cinnabookkeeping.domain

object CategoryCatalog {
    val expenses = listOf(
        BookkeepingCategory(RecordKind.EXPENSE, "餐饮", "🍜"),
        BookkeepingCategory(RecordKind.EXPENSE, "交通", "🚌"),
        BookkeepingCategory(RecordKind.EXPENSE, "购物", "🛍️"),
        BookkeepingCategory(RecordKind.EXPENSE, "娱乐", "🎮"),
        BookkeepingCategory(RecordKind.EXPENSE, "住房", "🏠"),
        BookkeepingCategory(RecordKind.EXPENSE, "医疗", "💊"),
        BookkeepingCategory(RecordKind.EXPENSE, "教育", "📚"),
        BookkeepingCategory(RecordKind.EXPENSE, "运动", "🏃"),
        BookkeepingCategory(RecordKind.EXPENSE, "旅行", "✈️"),
        BookkeepingCategory(RecordKind.EXPENSE, "日用", "🧴"),
        BookkeepingCategory(RecordKind.EXPENSE, "宠物", "🐾"),
        BookkeepingCategory(RecordKind.EXPENSE, "其他", "📦")
    )

    val incomes = listOf(
        BookkeepingCategory(RecordKind.INCOME, "工资", "💼"),
        BookkeepingCategory(RecordKind.INCOME, "奖金", "🎁"),
        BookkeepingCategory(RecordKind.INCOME, "副业", "💡"),
        BookkeepingCategory(RecordKind.INCOME, "理财", "📈"),
        BookkeepingCategory(RecordKind.INCOME, "红包", "🧧"),
        BookkeepingCategory(RecordKind.INCOME, "其他", "📦")
    )

    fun categories(kind: RecordKind): List<BookkeepingCategory> =
        if (kind == RecordKind.EXPENSE) expenses else incomes

    fun emoji(category: String): String =
        (expenses + incomes).firstOrNull { it.name == category }?.emoji ?: "📦"
}
